controller/src/beerocks/master/tasks/client_steering_task.cpp

```diff
@@ -174,42 +175,51 @@ void client_steering_task::steer_sta()
                          &client_disallow_event);
     }
 
-    // Send disconnect (802.11v)
-    auto bss_steer_request =
-        message_com::create_vs_message<beerocks_message::cACTION_CONTROL_CLIENT_BSS_STEER_REQUEST>(
-            cmdu_tx, id);
-
-    if (bss_steer_request == nullptr) {
-        LOG(ERROR) << "Failed building ACTION_CONTROL_CLIENT_BSS_STEER_REQUEST message!";
+    // Send STEERING request
+    if (!cmdu_tx.create(0, ieee1905_1::eMessageType::CLIENT_STEERING_REQUEST_MESSAGE)) {
+        LOG(ERROR) << "cmdu creation of type CLIENT_STEERING_REQUEST_MESSAGE, has failed";
         return;
     }
-    bss_steer_request->params().mac            = network_utils::mac_from_string(sta_mac);
-    bss_steer_request->params().disassoc_timer = disassoc_timer;
-    bss_steer_request->params().bssid.mac      = network_utils::mac_from_string(hostap_mac);
-    bss_steer_request->params().bssid.channel  = database.get_node_channel(hostap_mac);
-
-    if (disassoc_imminent) {
-        TASK_LOG(DEBUG) << "diassoc_imminent enabled";
-        bss_steer_request->params().disassoc_imminent = 1;
-    } else {
-        TASK_LOG(DEBUG) << "diassoc_imminent disabled";
-        bss_steer_request->params().disassoc_imminent = 0;
+
+    auto steering_request_tlv = cmdu_tx.addClass<wfa_map::tlvSteeringRequest>();
+
+    if (!steering_request_tlv) {
+        LOG(ERROR) << "addClass wfa_map::tlvSteeringRequest failed";
+        return;
     }
 
     auto current_ap_mac = database.get_node_parent(sta_mac);
 
+    steering_request_tlv->request_flags().request_mode =
+        wfa_map::tlvSteeringRequest::REQUEST_IS_A_STEERING_MANDATE_TO_TRIGGER_STEERING;
+    steering_request_tlv->request_flags().btm_disassociation_imminent_bit = disassoc_imminent;
+
+    steering_request_tlv->btm_disassociation_timer() = disassoc_timer;
```

Comment:
This line is the main reason why I asked to revert this.
It is a pitfall.
`disassoc_timer` is defined by intel hardware in units of beacon intervals which is 100ms.
The standard, on the other hand, defines it in a TU unit which is approximately 1ms.
I think that the right thing to do is add "_ms" prefix to all the "disassoc_timer" references from controller to the ap_manager, including yaml files in common and framework, and convert to 100ms beacon interval in bwl function.

Reply:
> This line is the main reason why I asked to revert this.
> It is a pitfall.
> `disassoc_timer` is defined by intel hardware in units of beacon intervals which is 100ms.
> The standard, on the other hand, defines it in a TU unit which is approximately 1ms.

Yes, I created #323 for this.

Note that it is already wrong, at least on Turris on the dev/turris-omnia-support branch, because the number gets passed as-is to bwl, which passes it directly to the hostapd BSS_TM_REQ command to hostapd. hostapd expects a value in TU there (I checked in he hostapd source code). Note that AFAIU, the practical effect of the disassociation timer value in the BTM request is none, because the client will just steer immediately anyway.

> I think that the right thing to do is add "_ms" prefix to all the "disassoc_timer" references from controller to the ap_manager, including yaml files in common and framework, and convert to 100ms beacon interval in bwl function.

It should be a `_tu` suffix, to be precise. (For reference, a TU is 1024µs) Also, I would express everything in TU, because that's what most of the WFA standards do.

Reply:
> Note that it is already wrong, at least on Turris on the dev/turris-omnia-support branch, because the number gets passed as-is to bwl, which passes it directly to the hostapd BSS_TM_REQ command to hostapd. hostapd expects a value in TU there (I checked in he hostapd source code). 

This strengthens my point. The code in the controller and agent should be aligned to the standard units, and be converted on bwl, which is vendor-specific.

> It should be a _tu suffix, to be precise. (For reference, a TU is 1024µs) Also, I would express everything in TU, because that's what most of the WFA standards do.

That's fine by me, as long as the unit is explicitly mentioned in the variable name.
I prefered "ms" because everyone knows its meaning, but will google TU. 
But TU is also fine.

Reply:
Done. Please review.